Elevel/Application/Features/ApplicationUserFeatures/GetAllApplicationUserQuery.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            public string LastName { get; set; }
+            public string FirstName { get; set; }
+            public string Email { get; set; }
+        }
+
+        public class Handler : IRequestHandler<Request, Response>
+        {
+
+            private readonly IMapper _mapper;
+
+            private readonly IApplicationDbContext _context;
+
+            private readonly UserManager<ApplicationUser> _userManager;
+
+            public Handler(IMapper mapper, UserManager<ApplicationUser> userManager)
+            {
+                _mapper = mapper;
+                _userManager = userManager;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+
+                IQueryable<ApplicationUser> users;
+
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName
+                        && x.FirstName == request.FirstName
+                        && x.Email == request.Email);
+                }
+                else 
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName
+                        && x.FirstName == request.FirstName);
+                }
+                else 
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.LastName == request.LastName
+                        && x.Email == request.Email);
+                }
+                else 
+                if (string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
+                        x.FirstName == request.FirstName
+                        && x.Email == request.Email);
+                }
+                else 
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && string.IsNullOrWhiteSpace(request.FirstName) 
+                    && string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
```

Comment:
and it looks that you can easy simple this by
if (!string.IsNullOrWhiteSpace(request.LastName) 
{ users = users.Where(filter by lstname) ]
if (first name)
{ users = users.Where (filter by first name) and etc.